packages/vite/src/ember.ts

```diff
@@ -0,0 +1,52 @@
+import { templateTag } from './template-tag.js';
+import { resolver } from './resolver.js';
+import { mergeConfig, type UserConfig } from 'vite';
+import { esBuildResolver } from './esbuild-resolver.js';
+
+export let extensions = ['.mjs', '.gjs', '.js', '.mts', '.gts', '.ts', '.hbs', '.hbs.js', '.json'];
+
+export function ember() {
+  return [
+    templateTag(),
+    resolver(),
+    {
+      name: 'vite-plugin-ember-config',
+      async config(config: UserConfig, env: { mode: string; command: string }) {
+        if (config?.resolve?.extensions) {
+          if (extensions.some(ext => !config?.resolve?.extensions?.includes(ext))) {
+            throw new Error(
+              `You are customising your resolve.extensions but you are missign some that are required for Ember. Make sure you include at least all of the following ${extensions}`
```

Comment:
Could explain here they can import the extensions from `@embroider/vite` to spread into their array?

Reply:
so my thinking was that if someone was customising their extensions it might seem strange/magic to just add a bunch to it 🤔 but I convinced myself that it was probably ok when I was thinking about your comment, especially because things will just stop working if you don't 🤷 

I'll leave this comment unresolved in case someone else has an opinion but I have swapped it to just add to the extensions if they are configured 👍

Reply:
we simplified this a bit more again so that people would be able to **exclude** extensions if they wanted to 👍